skops/card/tests/test_card.py

```diff
@@ -145,6 +146,28 @@ def test_save_model_card(destination_path, model_card):
     assert (Path(destination_path) / "README.md").exists()
 
 
+def test_model_caching(
+    skops_model_card_metadata_from_config, iris_skops_file, destination_path
+):
+    # _load_model get called
+    card = Card(iris_skops_file, metadata=metadata_from_config(destination_path))
+    with mock.patch("skops.card._model_card._load_model") as mock_load_model:
+        model1 = card.get_model()
+        model2 = card.get_model()
+        assert model1 is model2
+        # model is cached, hence _load_model is not called
+        mock_load_model.assert_not_called()
+        # update card with new model
+        new_model = LogisticRegression()
+        _, save_file = save_model_to_file(new_model, ".skops")
+        del card.model
+        card.model = save_file
+        model3 = card.get_model()  # model gets cached
+        model4 = card.get_model()
+        assert model3 is model4
+        assert mock_load_model.call_count == 1
```

Comment:
I see the intent with this test, but I think it's problematic that `del card.model` and `card.model = save_file` are being used. As a skops user, I wouldn't do that and I would still expect the cached model loading to work correctly. Therefore, I made some changes to the test so that these lines are not needed:

```suggestion
    new_model = LogisticRegression(random_state=4321)
    # mock _load_model, it still loads the model but we can track call count
    mock_load_model = mock.Mock(side_effect=load)
    card = Card(iris_skops_file, metadata=metadata_from_config(destination_path))
    with mock.patch("skops.card._model_card._load_model", mock_load_model):
        model1 = card.get_model()
        model2 = card.get_model()
        assert model1 is model2
        # model is cached, hence _load_model is not called
        mock_load_model.assert_not_called()

        # override model with new model
        dump(new_model, card.model)

        model3 = card.get_model()
        assert mock_load_model.call_count == 1
        assert model3.random_state == 4321

        model4 = card.get_model()
        assert model3 is model4
        assert mock_load_model.call_count == 1  # cached call
```

(line 3: `load` needs to be imported from `skops.io`)

This test is similar to yours but is _closer_ to how a user would actually use the model card. Please take a look and see if you agree with me. It would also be good to have a comment at the start of the test to explain what is being tested here.

Reply:
I can see how the user would go with your approach first, rather than mine. Definitely, I agree with your suggestion.

Reply:
Tested your suggestion and it passes the test as expected. I also added a short comment to describe what the function tests at the beginning of it.